File: 151001/Maliauka/src/main/java/by/bsuir/dc/features/editor/dto/CreateEditorDto.java
package by.bsuir.dc.features.editor.dto;

import by.bsuir.dc.features.editor.Editor;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import java.io.Serializable;

/**
 * DTO for {@link Editor}
 */
public record CreateEditorDto(
        @NotBlank @Size(min = 2, max = 64) String login,
        @NotBlank @Size(min = 8, max = 128) String password,
        @NotBlank @Size(min = 2, max = 64) String firstName,
        @NotBlank @Size(min = 2, max = 64) String lastName
) implements Serializable {}

File: 151002/Shatko/publisher/src/modules/editors/dto/create-editor.dto.ts
import { createZodDto } from "nestjs-zod";
import { z } from "zod";

const Schema = z.object({
  login: z.string().min(2).max(64),
  password: z.string().min(8).max(128),
  firstname: z.string().min(2).max(64),
  lastname: z.string().min(2).max(64),
});

export class CreateEditorDto extends createZodDto(Schema) {}
